test/unified_format.py

```diff
@@ -1080,38 +1080,39 @@ def _testOperation_loop(self, spec):
         successes_key = spec.get('storeSuccessesAsEntity')
         iteration_key = spec.get('storeIterationsAsEntity')
         iteration_limiter_key = spec.get('numIterations')
-        if failure_key:
-            self.entity_map[failure_key] = []
-        if error_key:
-            self.entity_map[error_key] = []
-        if successes_key:
-            self.entity_map[successes_key] = 0
-        if iteration_key:
-            self.entity_map[iteration_key] = 0
+        for i in [failure_key, error_key]:
+            if i:
+                self.entity_map[i] = []
+        for i in [successes_key, iteration_key]:
+            if i:
+                self.entity_map[i] = 0
         i = 0
+        global IS_INTERRUPTED
         while True:
             if iteration_limiter_key and i >= iteration_limiter_key:
                 break
             i += 1
             if IS_INTERRUPTED:
                 break
             try:
+                if iteration_key:
+                    self.entity_map._entities[iteration_key] += 1
                 for op in spec["operations"]:
                     self.run_entity_operation(op)
                     if successes_key:
                         self.entity_map._entities[successes_key] += 1
-                if iteration_key:
-                    self.entity_map._entities[iteration_key] += 1
             except AssertionError as exc:
                 if failure_key or error_key:
                     self.entity_map[failure_key or error_key].append({
-                        "error": exc, "time": time.time()})
+                        "error": str(exc), "time": time.time(), "type": type(
+                            exc)})
```

Comment:
Are these keys defined somewhere? I'm wondering if it makes more sense to nest "type" under the "error" field instead of a new top level field.

Reply:
The spec says that the error should be the "textual description of the failure encountered.", which to me implies that it should be just a string, not a document. It also places no limits on additional top level fields.